linter/app/app.go

```diff
@@ -20,21 +20,29 @@ type Context struct {
 
 type Factories struct {
 	GeneratorFactory id.GeneratorFactory
-	CheckFactories   []spec.CheckFactory // All additional check factories to run
 }
 
 type Hooks struct {
+	// Parse specs
 	LoadSpecs func(specsDir string, logFunc func(string, ...interface{})) (spec.Specs, error)
+
+	// Get additional checks to run
+	GetCheckFactories func(specs spec.Specs) []spec.CheckFactory
```

Comment:
What if instead of directly storing various factories, the `Factories` struct has 2 funcs — `MakeIdGeneratorFactory` and `MakeCheckFactories` (and that second one can take in the specs as an argument). That's similar to what the RM does, and I think makes a bit more sense — when I think of a "hook" I expect it to _do_ something, not just create a factory.